soql-standalone-parser/src/main/scala/com/socrata/soql/parsing/RecursiveDescentParser.scala

```diff
@@ -0,0 +1,1521 @@
+
+  private sealed abstract class NullPlacement
+  private case object First extends NullPlacement
+  private case object Last extends NullPlacement
+
+  type Reader = ExtendedReader[Token]
+
+  trait ParseException extends Exception {
+    def position: Position = reader.first.position
+    val reader: Reader
+  }
+
+  private case class ParseResult[+T](reader: Reader, value: T) {
+    def map[U](f: T=>U): ParseResult[U] = copy(value = f(value))
+  }
+
+  // sigh... ut's unfortunate that these are all here rather than
+  // being close to the expectations they reflect, but alas Scala
+  // provides no way to create a nontrivial immutable value and have
+  // it cached other than by sticking it in a `val` on some object,
+  // and recreating these sets on-site has a fairly hefty runtime
+  // cost.
+  private[this] def s(xs: Expectation*) = ListSet(xs : _*)
+  private val ANDSET = s(AND())
+  private val LIMITOFFSETSET = s(LIMIT(), OFFSET())
+  private val LIMITSET = s(LIMIT())
+  private val OFFSETSET = s(OFFSET())
+  private val ORDERBYSEARCHSET = s(AnOrderBy, SEARCH())
+  private val ORDERBYSET = s(AnOrderBy)
+  private val SEARCHSET = s(SEARCH())
+  private val GROUPBYSET = s(AGroupBy)
+  private val WHERESET = s(WHERE())
+  private val HAVINGSET = s(HAVING())
+  private val TABLEIDCOLONSTARSET = s(ATableIdentifier, COLONSTAR())
+  private val LPARENSET = s(LPAREN())
+  private val COMMASET = s(COMMA())
+  private val TABLEIDSTARSET = s(ATableIdentifier, STAR())
+  private val ASSET = s(AS())
+  private val EXPRESSIONSET = s(AnExpression)
+  private val JOINSET = s(LEFT(), RIGHT(), FULL(), JOIN())
+  private val ASLPARENSET = s(AS(), LPAREN())
+  private val TABLEIDSET = s(ATableIdentifier)
+  private val QUERYPIPESET = s(QUERYPIPE())
+  private val ROWOPSET = s(QUERYPIPE(), QUERYUNION(), QUERYINTERSECT(), QUERYMINUS(), QUERYUNIONALL(), QUERYINTERSECTALL(), QUERYMINUSALL())
+  private val DISTINCTSET = s(DISTINCT())
+  private val FROMSET = s(FROM())
+  private val PARTITIONBYSET = s(APartitionBy)
+  private val ASCDESCSET = s(ASC(), DESC())
+  private val NULLNULLSSET = s(NULL(), NULLS())
+  private val RANGEROWSSET = s(RANGE(), ROWS())
+  private val OVERSET = s(OVER())
+  private val DOTLBRACKETSET = s(DOT(), LBRACKET())
+
+  // These are all collapsed into "an operator" for ease of
+  // interpretation by an end-user instead of getting a bunch of
+  // line-noise.  Leaving them here in case we want to bring them back
+  // for some reason...
+
+  // private val COLONCOLONSET = s(COLONCOLON())
+  // private val MINUSPLUSSET = s(MINUS(), PLUS())
+  // private val CARETSET = s(CARET())
+  // private val FACTORSET = s(STAR(), SLASH(), PERCENT())
+  // private val TERMSET = s(PLUS(), MINUS(), PIPEPIPE())
+  // private val COMPARISONSET = s(EQUALS(), LESSGREATER(), LESSTHAN(), LESSTHANOREQUALS(), GREATERTHAN(), GREATERTHANOREQUALS(), EQUALSEQUALS(), BANGEQUALS())
+  private val OPERATORSET = s(AnOperator)
```

Comment:
I don't know if I agree. Specific operators are relevant in specific cases, and the others are not relevant in those specific cases. Saying "an operator" suggests that any operator would work and that's not true.

But on the other hand, I'm unable to come up with an actual situation in which it doesn't make sense to say "an operator". (If you can, that would be interesting.) (And for the problem of ComparisonSet specifically, if we want it back, I'd suggest collapsing it to "a comparison operator" or some such.)